sdk/machinelearningservices/Azure.ResourceManager.MachineLearning/src/Generated/Models/ManagedIdentityCredential.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.MachineLearning.Models
+{
+    /// <summary> Credential for user managed identity. </summary>
+    public partial class ManagedIdentityCredential : DataReferenceCredential
+    {
+        /// <summary> Initializes a new instance of <see cref="ManagedIdentityCredential"/>. </summary>
+        internal ManagedIdentityCredential()
+        {
+            CredentialType = DataReferenceCredentialType.ManagedIdentity;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="ManagedIdentityCredential"/>. </summary>
+        /// <param name="credentialType"> [Required] Credential type used to authentication with storage. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        /// <param name="managedIdentityType"> ManagedIdentityCredential identity type. </param>
+        /// <param name="userManagedIdentityClientId"> ClientId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null. </param>
+        /// <param name="userManagedIdentityPrincipalId"> PrincipalId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null. </param>
+        /// <param name="userManagedIdentityResourceId"> Full arm scope for the Id. For ManagedIdentityType = SystemManaged, this field is null. </param>
+        /// <param name="userManagedIdentityTenantId"> TenantId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null. </param>
+        internal ManagedIdentityCredential(DataReferenceCredentialType credentialType, IDictionary<string, BinaryData> serializedAdditionalRawData, string managedIdentityType, string userManagedIdentityClientId, string userManagedIdentityPrincipalId, string userManagedIdentityResourceId, string userManagedIdentityTenantId) : base(credentialType, serializedAdditionalRawData)
+        {
+            ManagedIdentityType = managedIdentityType;
+            UserManagedIdentityClientId = userManagedIdentityClientId;
+            UserManagedIdentityPrincipalId = userManagedIdentityPrincipalId;
+            UserManagedIdentityResourceId = userManagedIdentityResourceId;
+            UserManagedIdentityTenantId = userManagedIdentityTenantId;
+            CredentialType = credentialType;
+        }
+
+        /// <summary> ManagedIdentityCredential identity type. </summary>
+        public string ManagedIdentityType { get; }
+        /// <summary> ClientId for the UAMI. For ManagedIdentityType = SystemManaged, this field is null. </summary>
+        public string UserManagedIdentityClientId { get; }
```

Comment:
What's the format of those ID properties?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
I don't want to enforce is from SDK when API is allowing any string.
https://msdata.visualstudio.com/Vienna/_git/vienna?path=/src/azureml-api/src/ManagementFrontEnd/Contracts/V20240701Preview/Assets/DataReferenceCredential.cs&_a=contents&version=GBmaster